### PR TITLE
Apptesting: Change Interface to Struct

### PR DESCRIPTION
**Pull request by mattverse**
## Description

Extension for pull https://github.com/osmosis-labs/osmosis/pull/1008.
We should compare this with the interface approach and compare on usability
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by mattverse** [comment]
@ValarDragon this is just a draft as of now, can you review and see if this was similar to the approach you were thinking of?

File: x/superfluid/keeper/keeper_test.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 
 type KeeperTestSuite struct {
 	suite.Suite
+	helper apptesting.KeeperTestHelper
```
Comment (ValarDragon) [inline x/superfluid/keeper/keeper_test.go:30]: Can we remove the embedded `suite.Suite` and just make the `apptesting.KeeperTestHelper` embedded here?

File: app/apptesting/test_suite.go [hunk 2]
```diff
@@ -19,70 +19,70 @@ import (
...
 )
 
-// TODO: Consider an embedded struct here rather than an interface
-type SuiteI interface {
-	GetSuite() *suite.Suite
-	GetCtx() sdk.Context
-	SetCtx(sdk.Context)
-	GetApp() *app.OsmosisApp
+type KeeperTestHelper struct {
+	Suite suite.Suite
```
Comment (ValarDragon) [inline app/apptesting/test_suite.go:23]: Can we make the suite.Suite embedded here?

File: x/superfluid/keeper/keeper_test.go (hunk 1 above)
Comment (ValarDragon) [reply x/superfluid/keeper/keeper_test.go:30]: (or try how that makes things)

Comment (mattverse) [reply x/superfluid/keeper/keeper_test.go:30]: gotcha, I like that approach as well!

**Review by ValarDragon — APPROVED** [review]
LGTM! I like the struct approach much more than the initial interface idea

**Pull request merged**